opentracing-metrics-micrometer/src/test/java/io/opentracing/contrib/metrics/micrometer/MicrometerMetricsReporterTest.java

```diff
@@ -0,0 +1,182 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.metrics.micrometer;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.opentracing.contrib.api.SpanData;
+import io.opentracing.contrib.metrics.MetricLabel;
+import io.opentracing.contrib.metrics.label.BaggageMetricLabel;
+import io.opentracing.tag.Tags;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class MicrometerMetricsReporterTest {
+
+    public static final String METRIC_LABEL_NAME = "foo";
+    public static final String METRIC_LABEL_VALUE = "bar";
+    public static final String BAGGAGE_LABEL_NAME = "transaction";
+    public static final String BAGGAGE_LABEL_VALUE = "n/a";
+    public static final String TAG_LABEL_NAME = "Tag";
+    public static final String TAG_LABEL_VALUE = "IAmATag";
+
+    @Before
+    public void init() {
+        for (MeterRegistry registry : Metrics.globalRegistry.getRegistries()) {
+            Metrics.removeRegistry(registry);
+        }
+        Metrics.addRegistry(new SimpleMeterRegistry());
+    }
+
+    @Test
+    public void testWithCustomMetricTypeNames() {
+        // prepare
+        MicrometerMetricsReporter reporter = MicrometerMetricsReporter.newMetricsReporter()
+                .withName("MyName")
+                .withConstLabel("span.kind", Tags.SPAN_KIND_CLIENT) // Override the default, to make sure span metrics reported
+                .build();
+
+        SpanData spanData = mock(SpanData.class);
+        when(spanData.getOperationName()).thenReturn("testop");
+        when(spanData.getTags()).thenReturn(Collections.<String,Object>emptyMap());
+        when(spanData.getDuration()).thenReturn(100000L);
+
+        // test
+        reporter.reportSpan(spanData);
+
+        // verify
+        List<String> ltags = new ArrayList<>(6);
+        ltags.add("error");ltags.add("false");
+        ltags.add("operation");ltags.add("testop");
+        ltags.add("span.kind");ltags.add(Tags.SPAN_KIND_CLIENT);
+        String[] tags = ltags.toArray(new String[6]);
+
+        assertEquals(100000L,
+                (long) Metrics.timer("MyName", tags).totalTime(TimeUnit.MICROSECONDS)
+        );
+        assertEquals(1, Metrics.timer("MyName", tags).count());
+    }
+
+    @Test
+    public void testWithTagAndBaggageLabels() {
```

Comment:
This test is showing use of the default values for the additional tag and baggage labels, which is fine. But we also need a version of the test that shows the values being overridden in the span.